src/psyclone/psyGen.py

```diff
@@ -4657,17 +4668,72 @@ def apply(self):
 
 class IfBlock(Node):
     '''
-    Class representing an if-block within the PSyIR.
+    Class representing an if-block within the PSyIR. It has two mandatory
+    children: the first one represented the if-condition and the second one
+    the if-body; and an optional third child representing the else-body.
 
     :param parent: the parent of this node within the PSyIR tree.
     :type parent: :py:class:`psyclone.psyGen.Node`
+    :param str annotation: Tags that provide additional information about \
+        the node. The node should still be functionally correct when \
+        ignoring this tags.
+    :raises InternalError: when initialised with invalid parameters.
     '''
-    def __init__(self, parent=None):
+    valid_annotations = ('was_elseif', 'was_single_stmt', 'was_case')
+
+    def __init__(self, parent=None, annotation=None):
         super(IfBlock, self).__init__(parent=parent)
-        self._condition = ""
+        if annotation in IfBlock.valid_annotations:
+            self._annotations.append(annotation)
+        elif annotation:
+            raise InternalError(
+                "IfBlock with unrecognized annotation '{0}', valid annotations"
+                " are: {1}.".format(annotation, IfBlock.valid_annotations))
 
-    def __str__(self):
-        return "If-block: "+self._condition
+    @property
+    def condition(self):
+        ''' Return the PSyIR Node representing the conditional expression
+        of this IfBlock.
+
+        :return: IfBlock conditional expression.
+        :rtype: :py:class:`psyclone.psyGen.Node`
+        :raises InternalError: If the condition is not attached to the tree.
+        '''
+        if len(self.children) < 1:
+            raise InternalError(
+                "IfBlock malformed or incomplete. It should have at least 1 "
+                "children, but found {0}.".format(len(self.children)))
```

Comment:
Updated to 2, I had 1 because you can still get the condition of a partially malformed IF (just one child) but it is probably more robust to always check it is fully formed.